### PR TITLE
Add set/clear sources and boundary conditions methods to Python interface

### PR DESCRIPTION
**Pull request by quocdang1998**
This PR adds the methods setting/clearing sources and boundary conditions methods to Python interface.

Note: boundary conditions and sources arguments in ``OptionBlock`` to be removed (with regression tests to be updated).

### CODE REVIEW

File: python/lib/solver.cc [hunk 1]
```diff
@@ -418,6 +418,67 @@ WrapLBS(py::module& slv)
...
+    py::arg("volumetric_source")
+  );
+  lbs_problem.def(
+    "ClearBoundaries",
+    &LBSProblem::ClearBoundaries,
+    R"(
+    Clears all the boundary conditions from the solver.
+    )"
+  );
+  lbs_problem.def(
```
Comment (wdhawkins) [inline python/lib/solver.cc:458]: Instead of `SetBoundaryOptions`, can we call this `SetBoundaries`?

**Comment by wdhawkins** [comment]
Before this is committed, the current processing that is done in `SetOptions` needs to be removed, and the regression tests using the functionality need to be updates to use the new methods.

**Comment by wdhawkins** [comment]
It looks like the boundary/source processing that is done in `SetOptions` still needs to be removed. Users should have to use the new `Set` routines instead of `SetOptions` (this applies to `lbs_problem.cc` and `python/lib/solver.cc`).
